randovania/game_connection/executor/dread_executor.py

```diff
@@ -305,12 +315,16 @@ async def _send_keep_alive(self) -> bytes:
                 )
                 self.disconnect()
 
-    async def run_lua_code(self, current: str):
+    async def run_lua_code(self, current: str) -> None:
+        if self._socket is None:
+            return None
         self.code = current
         self._socket.writer.write(self._build_packet(PacketType.PACKET_REMOTE_LUA_EXEC, current.encode("utf-8")))
         await asyncio.wait_for(self._socket.writer.drain(), timeout=30)
 
-    async def bootstrap(self):
+    async def bootstrap(self) -> None:
+        if self._socket is None:
```

Comment:
wouldn't it be better to assert? not sure of the cases this function is called,

Reply:
Looks like its only getting called at the beginning when trying to connect. So somehow the socket would need to get reset in between RDV trying to connect and RDV sending the bootstrap to the game.